src/nccl_ofi_net.c

```diff
@@ -1294,6 +1373,44 @@ static ncclResult_t ofi_init(ncclDebugLogger_t logFunction)
 			       ofi_info_list->fabric_attr->prov_name);
 	}
 
+	/* Check if provider uses remote virtual addressing */
+	if (ofi_info_list->domain_attr->mr_mode & FI_MR_VIRT_ADDR) {
+		NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Provider %s uses remote virtual addressing",
+			       ofi_info_list->fabric_attr->prov_name);
+		virt_addr_mr = true;
+	} else {
+		NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Provider %s does not use remote virtual addressing",
+			       ofi_info_list->fabric_attr->prov_name);
+	}
+
+	/* Check if provider uses selects memory regisrtation keys */
+	if (ofi_info_list->domain_attr->mr_mode & FI_MR_PROV_KEY) {
```

Comment:
I am not sure how this works.

`FI_MR_PROV_KEY` was specified in hints->domain_attr->mr_mode.

hints was used to select a libfabric provider. 

The provider selection requires  hints's `mr_mode` to match provider's capability.

If a provider does not support `FI_MR_PROV_KEY`, the provider will not be selected in the first place.

Reply:
I see. You are right.

Reply:
Just a side note.

It is actually quite easy for a libfabric provider to implement the FI_MR_PROV_KEY capablity.

The provider just need to use the `ofi_mr_map` structure, the code is at 


https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_mr_map.c#L58

Reply:
Not having `MR_PROV_KEY` is actually intended to be a feature. It allows the application to choose the key and this can avoid overheads from key exchange and key table lookups. Of course, if you don't have that usage model, it's kind of an anti-feature. 😄